skfp/preprocessing/filters/rule_of_4.py

```diff
@@ -0,0 +1,91 @@
+from typing import Optional
+
+from rdkit.Chem import Mol
+from rdkit.Chem.Crippen import MolLogP
+from rdkit.Chem.Descriptors import MolWt
+from rdkit.Chem.rdMolDescriptors import CalcNumHBA, CalcNumRings
+
+from skfp.bases.base_filter import BaseFilter
+
+
+class RuleOf4(BaseFilter):
+    """
+    Rule designed to look for molecules used as PPI (protein-protein inhibitor).
+    Described in [1].
+
+    Acording to this rule PPI compound should meet following criteria:
+    - molecular weight >= 400 daltons
+    - HBA >= 4
+    - logp >=4
+    - number of rings >= 3
```

Comment:
Refactor similarly to previous comments. Also, logp -> logP

Reply:
Done